server/chat/backend/agent/tools/cloud_tools.py

```
@@ -686,45 +686,52 @@ def wrapper(*args, **kwargs):
     def wrapper(*args, **kwargs):
         # ALWAYS get user_id from context, never from AI
         context = get_user_context()
         context_user_id = context.get('user_id') if isinstance(context, dict) else context
         if context_user_id:
-            kwargs['user_id'] = context_user_id
-            logging.info(f"with_forced_context: Forced user_id from context: {context_user_id} for {func.__name__}")
+            if 'user_id' in func_params:
+                kwargs['user_id'] = context_user_id
+                logging.info(f"with_forced_context: Forced user_id from context: {context_user_id} for {func.__name__}")
         else:
             logging.error(f"with_forced_context: No user_id in context for {func.__name__}")
```

Ruff (0.15.17)

[warning] Logging statement uses f-string

(G004)